.gitlab/docker/Dockerfile.spack_compiler

```diff
@@ -20,6 +21,13 @@ RUN spack compiler info $COMPILER > /dev/null 2> /dev/null; compiler_missing=$?;
     if [[ $compiler_missing != 0 ]]; then \
         spack install $spack_compiler arch=$ARCH && \
         spack compiler add --scope site $(spack location -i $spack_compiler) && \
+        if [[ ! -z "$NVHPC_COMPILER" ]]; then \
+            nvhpc_version=${NVHPC_COMPILER#nvhpc@} && \
+            spack install $NVHPC_COMPILER%$spack_compiler arch=$ARCH && \
+            spack compiler rm gcc@11 && \
```

Comment:
I don't particularly love this since it's specific to the gcc that happens to be installed in the base image.

As an alternative, could you try a combination of the below:
- Use `--fresh` for the pika environment concretization, since this should end up preferring the compiler(s) you specify in the spec
- Add `COMPILER` and `NVHPC_COMPILER` to package preferences in the spack configuration. These should then be preferred to the bootstrap compiler.

Reply:
I meant to comment this earlier but forgot: note also that we don't mind if `build`-only dependencies (like `ninja`) are using the system compiler.